projects/hyperdash-angular/src/configuration/default-configuration.service.ts

```diff
@@ -1,5 +1,5 @@
-import { AbstractType, Inject, Injectable, Injector, Type } from '@angular/core';
-import { Deserializer, ModelPropertyTypeRegistrationInformation } from '@hypertrace/hyperdash';
+import { type AbstractType, Inject, Injectable, Injector, type Type } from '@angular/core';
```

Comment:
So we're requiring type imports now? I like that, but that will cause a lot of code changes - probably would make more sense to turn new linters like that on in separate PRs after migrating existing.

Reply:
```
    "@typescript-eslint/consistent-type-imports"
```
Yeah, the above rule is imposing this. It wasn't too bad to migrate but I can disable it for ht-ui.

Reply:
It makes it way slower to migrate and harder to meaningfully review though - that's why we don't add functionality while migrating :(

Reply:
Sure. Because we are migrating from tslint to eslint, the rule diff is a lot. Finding an exact match for each rule and migrating code is a cumbersome process. So, I used the OOTB migration script and it automatically came up with new rules based on our old rules and the recommended standards. 

What would you recommend? My plan is to verify the rules in hyperdash angular and finalize this set here. Then use the same set in other repos and migrate everything in one PR.

Reply:
> What would you recommend? 

My recommendation would be to use that migration script as a starting point, then see what code changes are required and find the appropriate rules. For example, this type change was a miss on the migrator, since that wasn't a rule we had previously (since the type import didn't exist in tslint's lifetime!). For those rules that are causing code changes, fix them up by hand to match existing behavior.

There may be some code changes required due to things that don't map cleanly or had bugs etc., but in the migration PR we should try for as few code changes - other than linter directive rewrites - as possible.

> My plan is to verify the rules in hyperdash angular and finalize this set here. Then use the same set in other repos

I would then use the result of above for other repos, but start a discussion on the newly supported rules or any changes we might want to make (since it's been a long time since we've last had that, anyway). We can go through those changes one by one and as we reach agreement on each, put it in its own PR and propagate the new/updated rule across repos.